apps/client/src/features/rundown/Rundown.tsx

```diff
@@ -16,6 +17,18 @@ import style from './Rundown.module.scss';
 
 const RundownEntry = lazy(() => import('./RundownEntry'));
 
+interface EventIdSwappingStore {
+  event: { index: number | null; id: string | null };
+  setEvent: (newEventId: string | null, newEventIndex: number | null) => void;
+  clearEvent: () => void;
+}
+
+export const useEventIdSwapping = create<EventIdSwappingStore>((set) => ({
+  event: { index: null, id: null },
+  setEvent: (newEventId, newEventIndex) => set(() => ({ event: { index: newEventIndex, id: newEventId } })),
+  clearEvent: () => set(() => ({ event: { index: null, id: null } })),
+}));
+
```

Comment:
This component and its near tree have admittedly confusing scope. 
Perhaps we should extract this into its own file?

Reply:
We should. Maybe in a the `rundown` folder?

Reply:
That sounds good to me